website/docs/r/nsxt_alb_service_engine_group.html.markdown

```diff
@@ -0,0 +1,101 @@
+---
+layout: "vcd"
+page_title: "VMware Cloud Director: vcd_nsxt_alb_service_engine_group"
+sidebar_current: "docs-vcd-resource-nsxt-alb-service-engine-group"
+description: |-
+  Provides a resource to manage NSX-T ALB Service Engine Groups. A Service Engine Group is an isolation domain that also
+  defines shared service engine properties, such as size, network access, and failover. Resources in a service engine
+  group can be used for different virtual services, depending on your tenant needs. These resources cannot be shared
+  between different service engine groups.
+---
+
+# vcd\_nsxt\_alb\_service\_engine\_group
+
+Supported in provider *v3.4+* and VCD 10.2+ with NSX-T and ALB.
+
+Provides a resource to manage NSX-T ALB Service Engine Groups. A Service Engine Group is an isolation domain that also
+defines shared service engine properties, such as size, network access, and failover. Resources in a service engine
+group can be used for different virtual services, depending on your tenant needs. These resources cannot be shared
+between different service engine groups.
+
+~> Only `System Administrator` can create this resource.
+
+## Example Usage (Adding NSX-T ALB Service Engine Group)
+
+```hcl
+resource "vcd_nsxt_alb_controller" "first" {
+  name         = "aviController1"
+  description  = "first alb controller"
+  url          = "https://controller.myXZ"
+  username     = "admin"
+  password     = "Welcome@1234"
+  license_type = "ENTERPRISE"
+}
+
+data "vcd_nsxt_alb_importable_cloud" "cld" {
+  name          = "importable-cloud-name"
+  controller_id = vcd_nsxt_alb_controller.first.id
+}
+
+resource "vcd_nsxt_alb_cloud" "first" {
+  name        = "nsxt-cloud"
+  description = "first alb cloud"
+
+  controller_id       = vcd_nsxt_alb_controller.first.id
+  importable_cloud_id = data.vcd_nsxt_alb_importable_cloud.cld.id
+  network_pool_id     = data.vcd_nsxt_alb_importable_cloud.cld.network_pool_id
+}
+
+resource "vcd_nsxt_alb_service_engine_group" "first" {
+  name                      = "demo-service-engine"
+  description               = "Service Engine for Terraform documentation"
+  alb_cloud_id              = vcd_nsxt_alb_cloud.first.id
+  service_engine_group_name = "Default-Group"
+  reservation_model         = "SHARED"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) A name for NSX-T ALB Service Engine Group
+* `description` - (Optional) An optional description NSX-T ALB Service Engine Group
+* `alb_cloud_id` - (Required) A reference NSX-T ALB Cloud. Can be looked up using `vcd_nsxt_alb_cloud` entity
+* `reservation_model` - (Required) Definition if the Service Engine Group is `DEDICATED` or `SHARED`
+* `service_engine_group_name` - (Required) Name of available Service Engine Group in ALB
+
+~> There is a special argument `sync_on_refresh` which is not passed to VCD, but alters behaviour of this resource so
```

Comment:
See annotation on the data source docs

Reply:
Fair point. Changed in both - resource and data source.